### ISSUE TITLE
I am very interested in your project. Is there any way I can get involved？:)

### ISSUE THREAD

**Issue by SophusDavid** [jake-austin/zipnerf-nerfstudio#1]

**Comment by jake-austin** [comment]
Sure thing! Feel free to make a branch and make a commit, terrance and I are trying to get this paper implemented and added as an external method for nerfstudio, with the later step of getting some of the components like the blurred proposal loss integrated into nerfstudio after we get all the bugs worked out here. Haven't made an open source repo like this before, so if there's an issue of permissions or anything like that lmk and I can make sure you have access that you need. 

The attempts rn are to port over some of the code from here https://github.com/SuLvXiangXin/zipnerf-pytorch to be nerfstudio, I am using their proposal loss and it seemed to work fine with nerfacto. I'm trying to iron out some of the bugs with multisampling at the moment (the code is still in a messy state from me debugging multisampling... oops hehe). 

There are some fundamental limitations we might come up against like batchsize and MLP size. Batchsize for the paper is 2^16, which is impossible for one GPU (at least for my GPUs lol), so I'm mimicing that by scaling down the batch size and learning rate accordingly. For MLP size, tinycudann doesn't allow hidden sizes greater than 128 so we're also fundamentally limited there unless we want to use torch MLPs which probably are ok, but tinycudann is faster and lighter

Again, feel free to make a branch and commit, keep us updated on your progress!

**Comment by Bin-ze** [comment]
Hello, I am also paying attention to the work on zipnerf, and then I saw your project, I tried to use it for training, but I still can't get satisfactory results.
Can you tell me the parts that have been realized so far, and the difficulties that have not been realized, I also want to participate in this discussion

**Comment by jake-austin** [comment]
Oh yeah just to clarify: this method is not working at this point... like I couldn't get any results at all that weren't noise, let alone results that are just less than satisfactory. We're still implementing it and ironing out all the bugs. I just wanted to get a repo that terrance and I could start working on, made it public for no reason in particular, and was unprepared for any external collaborators, but the more the merrier and I'm excited that people want to hop in, and I want to make sure you can help if this is something you're interested in! I have my best interpretation of multisampling/downweighting implemented here and clearly I did something wrong, but feel free to give it a spin or use the repo linked above as a starting point. I'm not sure what the best way to keep everyone in the loop on this, but I think it will probably happen through github issues and PRs. I might make one main github issue for general progress that ppl can report in which case I'll lyk here, but odds are it's just decentralized so just check github issues and any pending PRs for progress updates. Feel free to open up some PRs if you think you have a lead on debugging this, and I can take a look at them and merge them in

**Comment by SophusDavid** [comment]
Okay! I think i should start from understanding [the pytorch implementation ](https://github.com/SuLvXiangXin/zipnerf-pytorch) first.Thank u for your reply!I think we can keep this issue for a further disscussion.

**Comment by jake-austin** [comment]
I'll also update the main README with progress updates, feel free to add your own progress updates on the README when you make a commit just for a centralized TLDR of where we're at rn
